grapple/middleware.py

```diff
@@ -1,26 +1,11 @@
-import inspect
-
 from graphene import ResolveInfo
-from graphql.execution.middleware import MIDDLEWARE_RESOLVER_FUNCTION
+from graphql.execution.middleware import get_middleware_resolvers
 
 from .registry import registry
 
 ROOT_TYPES = ["Query", "Mutation", "Subscription"]
 
 
-def get_middleware_resolvers(middlewares):
-    for middleware in middlewares:
-        if inspect.isfunction(middleware):
-            yield middleware
-        if not hasattr(middleware, MIDDLEWARE_RESOLVER_FUNCTION):
-            raise Exception(
-                "Middleware must be either a class or a function. Got: {}.\nYou can read more about middleware here: https://docs.graphene-python.org/en/latest/execution/middleware/".format(
-                    type(middleware)
-                )
-            )
-        yield getattr(middleware(), MIDDLEWARE_RESOLVER_FUNCTION)
```

Comment:
Now it makes sense. This bit here instantiates the middleware class and calls the resolver, whereas https://github.com/graphql-python/graphql-core/blob/main/src/graphql/execution/middleware.py#L67 doesn't

will refresh my memory how we pass on the middleware down the chain and see if there's something we can do without folks outright instantiating the middleware classes

Reply:
@dopry ok, I needed more of a break than I thought I did :)

got a bit of time to dig into the code a bit more. we can solve this in https://github.com/torchbox/wagtail-grapple/blob/65a26a9daf4f7a3e937f7eaed90cd8373951be57/grapple/helpers.py#L40-L50 with a `if isinstance(middleware, type)` check.

Will follow this up